### PR TITLE
[Feature to main] Explainability in hybrid query

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
Merge "Explainability in hybrid query" from feature branch to main, we do have clearance from app sec team.

Includes PRs:
- https://github.com/opensearch-project/neural-search/pull/970

### Related Issues
Resolves https://github.com/opensearch-project/neural-search/issues/658

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [X] Commits are signed per the DCO using `--signoff`.
- [X] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/neural-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by heemin32 — APPROVED** [review]

**Review by ryanbogan — APPROVED** [review]
LGTM!  Once this is merged, we should try to add more tests and increase the code coverage.

**Comment by martin-gaievski** [comment]
> LGTM! Once this is merged, we should try to add more tests and increase the code coverage.

yup, for sure. Codecov was broken at time I was working on implementation PR, in this PR I just want to merge from feature branch, and anything else will be in a separate PR

**Review by navneet1v — APPROVED** [review]

**Pull request merged**
